client/ayon_core/tools/loader/control.py

```diff
@@ -348,10 +348,18 @@ def get_loaded_product_ids(self):
             return set()
 
         if not self._loaded_products_cache.is_valid:
-            if isinstance(self._host, ILoadHost):
-                containers = self._host.get_containers()
-            else:
-                containers = self._host.ls()
+            try:
+                if isinstance(self._host, ILoadHost):
+                    containers = self._host.get_containers()
+                else:
+                    containers = self._host.ls()
+
+            except BaseException:
+                self.log.error(
+                    "Falied to collect loaded products.", exc_info=True
```

Comment:
Typo:
```suggestion
                    "Failed to collect loaded products.", exc_info=True
```

I'm also wondering whether mentioning "containers" instead of "products" may be clearer here?
```suggestion
                    "Failed to collect loaded containers.", exc_info=True
```

Anyway, pick your juice.